lib/srv/db/common/autousers.go

```diff
@@ -51,7 +51,7 @@ type UserProvisioner struct {
 // database has been established to release the cluster lock acquired by this
 // function to make sure no 2 processes run user activation simultaneously.
 func (a *UserProvisioner) Activate(ctx context.Context, sessionCtx *Session) (func(), error) {
-	if !sessionCtx.AutoCreateUser {
+	if !services.IsCreateDatabaseUserEnabled(sessionCtx.AutoCreateUserMode) {
```

Comment:
I would throw errors if the Backend doesn't support the new mode yet (somewhere here). IMO, it's better to alert the user the mode configuration is wrong than silently falling back to another mode. But certainly up for discussion.